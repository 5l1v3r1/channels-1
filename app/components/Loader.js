// @flow
import React from 'react';
import { IconSpinner } from './Icons';
import styles from './Loader.scss';

const Loader = () => (
  <IconSpinner styles={styles} />
);

export default Loader;
