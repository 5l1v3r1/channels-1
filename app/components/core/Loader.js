// @flow
import React from 'react';
import { IconSpinner } from './Icons';
import styles from './Loader.scss';

const Loader = ({
  className
}: {
  className?: string
}) => (
  <div className={[styles.loader, className].join(' ')}>
    <IconSpinner className={styles.iconSpinner} />
  </div>
);

export default Loader;
