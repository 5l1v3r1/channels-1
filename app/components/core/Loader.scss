@import '../../app.global.scss';

.loader {
  position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	width: 40px;
	height: 40px;
}

.iconSpinner {
	width: 100%;
	height: 100%;
  fill: $red;
  animation: spin 0.6s linear infinite;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
