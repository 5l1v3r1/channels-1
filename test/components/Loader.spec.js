import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loader from '../../app/components/core/Loader';

describe('Loader', () => {
  it('should render self', () => {
    const enzymeWrapper = shallow(<Loader />);
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });
});
