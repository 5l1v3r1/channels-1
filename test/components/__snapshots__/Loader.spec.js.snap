// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render self 1`] = `
<div
  className="loader"
>
  <Component
    className="iconSpinner"
  />
</div>
`;
